ext/sqlite3/statement.c

```diff
@@ -72,7 +72,7 @@ prepare(VALUE self, VALUE db, VALUE sql)
     CHECK(db_ctx->db, status);
     timespecclear(&db_ctx->stmt_deadline);
 
-    return rb_str_new2(tail);
+    return rb_utf8_str_new_cstr(tail);
```

Comment:
Would it be better to reuse the encoding of the original string?

Reply:
Regardless of the original encoding, the query ends up encoded as UTF-8 in `Statement#initialize` and the remainder string being returned from `sqlite_prepare` is also UTF-8.

This call isn't re-encoding anything, it's just telling Ruby that the C string is in fact UTF-8 (the default is otherwise ASCII-8BIT which is what led to the problem you reported).

Reply:
Ah, I see. Alright, thanks for fixing this!